torch/nn/modules/module.py

```diff
@@ -297,7 +297,16 @@ def hammerblade(self):
         Returns:
             Module: self
         """
-        return self._apply(lambda t: t.hammerblade())
+        def apply_hb(t):
+            if isinstance(self, torch.nn.BatchNorm2d) and \
+                    (t.dtype == torch.int64):
+                # BatchNorm2d's batch counter is a long tensor. Since it is metadata
```

Comment:
is this the 64bit integer add we saw?

Reply:
Yes, later found out this not the core tensor in the kernel computation. This is just the metadata of `torch.nn.BatchNorm2d` layer. Meaning `torch.nn.functional.batch_norm` doesn't need this.